Core/Core/Use Cases/UploadManager.swift

```diff
@@ -190,60 +191,27 @@ open class UploadManager: NSObject, URLSessionDelegate, URLSessionTaskDelegate,
+    public func urlSession(_ session: URLSession, dataTask: URLSessionDataTask, didReceive data: Data) {
         Logger.shared.log()
-        let semaphore = DispatchSemaphore(value: 0)
-        var currentTask: URLSessionTask?
-        process.performExpiringActivity(withReason: "get file") { expired in
-            if expired {
-                currentTask?.cancel()
-            }
-            self.context.performAndWait {
-                guard let file = self.file(taskID: task.taskIdentifier) else { return }
-                if let response = task.response as? HTTPURLResponse,
-                    response.statusCode == 201,
-                    let location = response.allHeaderFields[HttpHeader.location] as? String,
-                    let url = URL(string: location),
-                    let user = file.user,
-                    let session = LoginSession.sessions.first(where: { user == $0 }) {
-                    let objectID = file.objectID
-                    var request = URLRequest(url: url)
-                    if let token = session.accessToken {
-                        request.setValue("Bearer \(token)", forHTTPHeaderField: HttpHeader.authorization)
-                    }
-                    request.setValue("application/json+canvas-string-ids", forHTTPHeaderField: HttpHeader.accept)
-                    currentTask = URLSession.getDefaultURLSession().dataTask(with: request) { data, _, error in
-                        self.context.performAndWait {
-                            defer { semaphore.signal() }
-                            guard let file = try? self.context.existingObject(with: objectID) as? File else { return }
-                            guard let data = data, error == nil else {
-                                self.complete(file: file, error: error ?? NSError.internalError())
-                                return
-                            }
-                            do {
-                                let result = try self.decoder.decode(APIFile.self, from: data)
-                                file.update(fromAPIModel: result)
-                                try self.context.save()
-                                if case let .submission(courseID, assignmentID, comment)? = file.context, error == nil {
-                                    self.submit(file: file, courseID: courseID, assignmentID: assignmentID, comment: comment)
-                                    return
-                                }
-                                self.complete(file: file, error: error)
-                            } catch {
-                                self.complete(file: file, error: error)
-                            }
-                        }
-                    }
-                    currentTask?.resume()
-                } else {
-                    self.complete(file: file, error: error)
-                    semaphore.signal()
-                }
```

Comment:
Enabling Background Modes fixed an issue that allowed me to remove all this stuff.